### ISSUE TITLE
xwayland: mouse input to sticky window ends up at any other xwayland window

### ISSUE THREAD

**Issue by David96** [swaywm/sway#7608]
- **Sway Version:**
  - sway version 1.9-dev-48d6eda3c (newest master, also of wlroots)

- **Description:**

I'm having the problem that the screen sharing controls of Zoom are not usable when any other xwayland window is open on the same screen.
The controls are displayed in a sticky floating xwayland window with the following properties:
```
WM_NAME(STRING) = "Zoom - Licensed Account"
_NET_WM_NAME(UTF8_STRING) = "Zoom - Licensed Account"
_MOTIF_WM_HINTS(_MOTIF_WM_HINTS) = 0x3, 0x3e, 0x6e, 0x0, 0x0
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_NORMAL
_NET_WM_STATE(ATOM) = _NET_WM_STATE_MAXIMIZED_VERT, _NET_WM_STATE_MAXIMIZED_HORZ
_XEMBED_INFO(_XEMBED_INFO) = 0x0, 0x1
WM_CLIENT_LEADER(WINDOW): window id # 0xe0001c
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		window id # of group leader: 0xe0001c
WM_CLIENT_MACHINE(STRING) = "***"
_NET_WM_PID(CARDINAL) = 135375
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 14680764
WM_CLASS(STRING) = "zoom ", "zoom"
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		user specified location: 0, 1250
		user specified size: 1920 by 1030
		program specified minimum size: 376 by 480
		window gravity: Static
```
Any mouse input on the controls will be passed to some other (random?) xwayland window. So for example if I have the mattermost client (xwayland electron app) open on any workspace (does not have to be the one currently shown!), any mouse clicks on the control window will be passed to the mattermost window. This only happens if the other xwayland window is on the same screen as the zoom control window. So to be able to interact with the zoom screen sharing controls, I have to either close all other xwayland windows or move them to another screen.

So in short, to reproduce:

1. Open a Zoom meeting in the Zoom App (using xwayland)
2. Share your screen
3. Notice that the screen sharing controls don't receive any input until you've closed all xwayland windows on the same screen.

Debug log of starting sway, starting zoom meeting, sharing screen, trying to click the controls, closing the zoom window (the only xwayland window) and then successfully stopping the screen share: 
[sway-debug.log](https://github.com/swaywm/sway/files/11565906/sway-debug.log)

**Comment by David96** [comment]
I was able to reproduce this issue using weston and therefore think it's an xwayland issue and not related to sway/wlroots. I reported it there: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1551

**Comment by David96** [comment]
Reopening since apparently this issue doesn't affect gnome, so it's probably not an xwayland issue but somehow coincidentally a bug that exists in weston and sway/wlroots?

**Comment by David96** [comment]
For more information see this comment on the xwayland issue: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1551#note_1935203
So this really seems to be an issue with sway getting the stacking order of the xwayland windows wrong.

**Comment by emersion** [comment]
I think this is related: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4052

**Comment by David96** [comment]
Thanks, I somehow missed this! I can confirm it fixes my issue but now sway crashes upon stopping the screen share, will post the stack trace to the linked merge request.

**Comment by smlx** [comment]
I guess this can be closed?
